lib/puppet/application/lookup.rb

```diff
@@ -331,6 +330,18 @@ def generate_scope
     end
 
     node = Puppet::Node.indirection.find(node) unless node.is_a?(Puppet::Node) # to allow unit tests to pass a node instance
+
+    if options[:fact_file]
+      original_facts = node.facts.values
+      if options[:fact_file].match(/.*\.json/)
```

Comment:
Better to use the `String.end_with?` method. (Faster, when not requiring a pattern based search, and does not require human interpretation of the regexp). (The regexp is not quite right BTW :-) since it will return a match for strings with ".json" anywhere inside). Just change to "end_with?" which is a string comparisson, which is wanted here anyway.